partiql-spi/src/main/java/org/partiql/spi/value/Datum.java

```diff
@@ -143,61 +142,72 @@ default byte getByte() {
     }
 
     /**
-     * @return the underlying value applicable to the types:
-     * {@link PartiQLValueType#DATE}.
-     * @throws UnsupportedOperationException if the operation is not applicable to the type returned from
-     *                                       {@link #getType()}; for example, if {@link #getType()} returns a {@link PartiQLValueType#INT}, then this method
-     *                                       will throw this exception upon invocation.
-     * @throws NullPointerException          if this instance also returns true on {@link #isNull()}; callers should check that
-     *                                       {@link #isNull()} returns false before attempting to invoke this method.
+     * @return a {@link LocalDate} for DATE, TIMESTAMP, and TIMESTAMPZ types.
```

Comment:
Non-blocking -- Given TIMESTAMP/Z has `getLocalDateTime`, this may be redundant. Same goes for `getLocalTime`.